client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx

```diff
@@ -215,8 +217,14 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow } ) => {
 			? isThemePurchased( state, selectedDesignThemeId, site.ID )
 			: false
 	);
+
+	const isPluginBundleEligible = useIsPluginBundleEligible();
+	const themeDetails = useThemeDetails( selectedDesign?.slug || '' );
```

Comment:
Sorry for jumping in. We (@Automattic/ganon) found that this will cause an unnecessary call to this `/themes/` endpoint on the first load, when no design is previewed yet (`selectedDesign` is undefined). Worse yet, since we pass an empty string, it will call the theme search endpoint, which will return a list of themes, instead of a single theme.

Can we please rethink this logic? One idea is to pass the `enabled` options to the react-query operation similar to the existing code:

https://github.com/Automattic/wp-calypso/blob/3f1c6193b530cedbda57c34ea80ff88139ff7a5b/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx#L148-L150

Context: we're investigating on how to optimize how the Design Picker loads (https://github.com/Automattic/wp-calypso/issues/67880), and just came across this new additional call in this PR.

Reply:
Also wasn't the plan to use this new field? D88069-code

Reply:
Yes! Here is the PR with the new field: https://github.com/Automattic/wp-calypso/pull/68361👍

Reply:
Thanks!